src/io/flutter/run/FlutterAppState.java

```diff
@@ -103,6 +105,13 @@ protected ProcessHandler startProcess() throws ExecutionException {
       }
     }
 
+    if (device != null) {
+      // Bring simulator to front.
+      if (device.emulator() && Objects.equals(device.platform(), "ios")) {
+        new OpenSimulatorAction(true).actionPerformed(null);
```

Comment:
Just to confirm, that re-issuing the open doesn't have side effects apart from bringing the app to the front?

One way that I've solved this in the past is to generate and execute an applescript on the fly which tells the app (specified by name) to come to the front.